### PR TITLE
bug 1319259: Update to node v6

### PR DESCRIPTION
**Pull request by jwhitlock**
* [x] Switch to node.js v6 for the KumaScript engine
* [x] Update ansible provisioning
* [x] Update docker
* [x] Update SCL3 deployment scripts
* [x] Switch to [mozmar/base](https://github.com/mozmar/docker-base)
* [ ] Update Jenkinsfile to build and upload the [kuma_base](https://quay.io/repository/mozmar/kuma_base) image
* [x] Merge KumaScript node v6 [PR 69](https://github.com/mozilla/kumascript/pull/69), switch in this PR.

### CODE REVIEW

**Comment by jwhitlock** [comment]
Here's what I did to run this locally in Docker (from memory, I may have a detail wrong):

```
docker build --tag kumascript:n6 kumascript
docker build --tag kuma_base:n6 --file Dockerfile-base .
```

In ``.env``, I added the line:
```
COMPOSE_FILE=docker-compose.yml:docker-compose.ks.yml
```

I created ``docker-compose.ks.yml`` with:
```
version: "2.1"
services:
  worker:
    image: kuma_base:n6
  web:
    image: kuma_base:n6
  api:
    image: kuma_base:n6
  kumascript:
    image: kumascript:n6
```
and then used it:
```
docker-compose down
make bash
# In web container
make build-static
make test
```

File: Dockerfile-base [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM quay.io/deis/base:0.3.0
+FROM quay.io/deis/base:v0.3.6
```
Comment (jgmize) [inline Dockerfile-base]: May want to switch this to quay.io/mozmar/base

Comment (jwhitlock) [inline Dockerfile-base]: Good idea. I'll move the deis v0.3.6 security update to a [mozmar/base](https://github.com/mozmar/docker-base) PR.

File: Jenkinsfile [hunk 2]
```diff
@@ -18,8 +18,9 @@ node {
           sh 'make compose-test'
         }
 
-        stage('Build & push kuma image') {
+        stage('Build & push images') {
           sh 'make build-kuma push-kuma'
+          sh 'make push-base VERSION=latest'
```
Comment (jwhitlock) [inline Jenkinsfile]: @jgmize I'm 50% sure this is wrong...

Maybe it push both ``kuma_base:<commit>`` and ``kuma_base:latest``:

```
sh 'make build-kuma push'
sh 'make push-base VERSION=latest'
```

Maybe branches should also push commit-tagged kuma_base images.

File: Dockerfile-base [hunk 3]
```diff
@@ -24,19 +23,27 @@ RUN apt-get update && \
...
+# Install the Node.js 6.9.2 LTS release.
+ENV DEB_PKG=nodejs_6.9.2-1nodesource1~jessie1_amd64.deb
+RUN curl -sLO https://deb.nodesource.com/node_6.x/pool/main/n/nodejs/${DEB_PKG}\
+    && dpkg -i ${DEB_PKG} \
+    && rm ${DEB_PKG} \
+    && apt-get install -f
+
 RUN update-alternatives --install /usr/bin/node node /usr/bin/nodejs 10
 
 RUN npm install -g \
```
Comment (escattone) [inline Dockerfile-base]: Add `node-sass@4.3.0` to the globally installed node packages to accommodate the upcoming change from stylus to sass.

Comment (jwhitlock) [inline Dockerfile-base]: I think that belongs in PR #4097, not this one.

Comment (escattone) [inline Dockerfile-base]: Yep, you're right, thanks.

**Pull request merged**